sdk/storage/Azure.Storage.Blobs.Cryptography/src/EncryptedBlobClient.cs

```diff
@@ -0,0 +1,323 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Security.Cryptography;
+using Azure.Core;
+using Azure.Core.Cryptography;
+using Azure.Core.Pipeline;
+using Azure.Storage.Blobs.Specialized.Cryptography.Models;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+
+namespace Azure.Storage.Blobs.Specialized.Cryptography
+{
+    /// <summary>
+    /// The <see cref="EncryptedBlobClient"/> allows you to manipulate
+    /// Azure Storage block blobs with client-side encryption. See
+    /// <see cref="BlobClient"/> for more details.
+    /// </summary>
+    public class EncryptedBlobClient : BlobClient
```

Comment:
I thought we were going to derive this from `BlockBlobClient` to leave future wiggle room?

Reply:
Discussed offline. Leaving as is.